### ISSUE TITLE
[DOC] Clarification on WKLoss

### ISSUE THREAD

**Issue by MohammadElSakka** [ayrna/dlordinal#133]
Hi, congratulations on the nice work. I would like a clarification on the WKLoss. 

In the [documentation](https://dlordinal.readthedocs.io/en/latest/losses.html#dlordinal.losses.WKLoss) you've given a formula for the WK Loss function that seems to be equal to $$1-k$$ where k is referenced in Equation (3) of the mentioned article[(Torre et al.)](https://www.sciencedirect.com/science/article/pii/S0167865517301666?casa_token=k7-ZDzutUesAAAAA:bQbc7OZGOwSRzgobHi1Zq5yB7UtSo3Xm9PboTXArCFloV1M3uncnQK882q46aPvHxvW3BmQLBMY).

To recall, the WK Loss that is to be *minimized* presented in the mentioned article looks like the following:

$$
\text{minimize} \quad \mathcal{L} = \log(1 - \kappa) \quad \text{where} \quad \mathcal{L} \in (-\infty, \log 2]
$$

So it seems to be that you removed the logarithm by applying the exponential function $e$, which in fact gives your version of the WK Loss:

$$
\text{minimize} \quad \mathcal{L} = 1 - \kappa\quad \text{where} \quad \mathcal{L} \in ]0, 2]
$$

So, I would like to understand why this logarithm has been removed? Thank you

**Comment by victormvy** [comment]
Hey, Mohammad.

Thank you! I am glad to see that you found our package useful.

Regarding your question about WKLoss: the authors in [Torre et al.](https://www.sciencedirect.com/science/article/pii/S0167865517301666?casa_token=k7-ZDzutUesAAAAA:bQbc7OZGOwSRzgobHi1Zq5yB7UtSo3Xm9PboTXArCFloV1M3uncnQK882q46aPvHxvW3BmQLBMY) defined

$$
\kappa = 1 - \frac{\mathcal{N}}{\mathcal{D}}, \quad \mathcal{N} \in \left[ 0, N \sum_{i=1}^{J} \sum_{j=1}^{J} \omega_{i,j} \right], \quad \kappa \in [0, 2]
$$

and they defined the loss as

$$
\mathcal{L} = \log(1 - \kappa), \quad \mathcal{L} \in [\textcolor{red}{-\infty}, \log 2]
$$

which can be rewritten as

$$
\mathcal{L} = \log(1 - (1 - \mathcal{N}/\mathcal{D})) =\log(\mathcal{N}/\mathcal{D})
$$

Since $0 \in \mathcal{N}$ (and therefore $\kappa = 1 - \mathcal{N}/\mathcal{D}$ can be 1), this can cause numerical instability due to the logarithm. For this reason, in [Vargas et al.](https://doi.org/10.1016/j.neucom.2020.03.034) we proposed to compute the loss function without the logarithm. This changes the range of the loss from $(-\infty, \log 2]$ to $[0, 2]$.

Since the function is a loss, this transformation does not affect its purpose and it effectively fixes the numerical stability issues.

We will also review the documentation to clarify this point, and we are considering adding an optional parameter in the future that allows users to compute the loss using the logarithm, consistent with the original proposal, while adding a small epsilon to maintain numerical stability.

**Comment by victormvy** [comment]
Removing the logarithm should not affect training, since the loss is still a function to minimize and is monotonically related to the original version. In fact, by improving numerical stability, it can sometimes work better in practice.  

Once we add the optional parameter that I mentioned, you will be able to directly compare both versions.

**Comment by MohammadElSakka** [comment]
Perfect thank you 👍
